master/main.go

```diff
@@ -236,16 +240,27 @@ func unregisterWorker(oldWorker *worker) {
 	if oldWorker.InstanceId != "" {
 		TerminateWorkers([]*worker{oldWorker})
 	}
+	//Redistribute graphs this worker was still processing
+	for _, task := range oldWorker.TasksProcessing {
+		distributeGraph(task.Graph, task.Parameters)
+	}
 }
 
 func distributeGraph(graph *graphs.Graph, parameters map[string][]string) {
 	var activeWorkers = getActiveWorkers()
-	// Distribute graph among workers by randomly selecting a worker
-	// possible improvement select the worker which has the shortest queue
-	var worker = activeWorkers[rand.Intn(len(activeWorkers))]
-	err := sendGraphToWorker(*graph, worker, parameters)
-	if err != nil {
+	for {
+		// Distribute graph among workers by randomly selecting a worker
+		// possible improvement select the worker which has the shortest queue
+		var worker = activeWorkers[rand.Intn(len(activeWorkers))]
+		err := sendGraphToWorker(*graph, worker, parameters)
+		if err == nil {
+			// TODO remove from this list again when processing is done by worker
```

Comment:
Nogsteeds een todo?

Reply:
ah ja thanks moet even iets maken zodat de worker laat weten aan de master dat die klaar is met een graph